src/librustdoc/html/static/js/main.js

```diff
@@ -134,6 +134,18 @@ function hideThemeButtonState() {
     themePicker.style.borderBottomLeftRadius = "3px";
 }
 
+function buildUrl(search, filterCrates) {
```

Comment:
Let's add some JSDoc comments to this function, indicating that filterCrates is optional - or better yet, make it non-optional, and specify that it should be "null" for "no filter". Then we can remove the `!== "undefined"` check below.

Also, let's move this function to `search.js`. It's search-specific functionality, after all. And we shouldn't call it in hideResults; we should keep the code currently there, which doesn't care about search URL parameters.

In general, many of the things in `searchState` can and should be moved to search.js eventually. What's in there is the stuff that had relationships with other things in main.js that I didn't want to untangle in my initial refactoring to split search.js.